homeassistant/components/balboa/entity.py

```diff
@@ -0,0 +1,57 @@
+"""Base class for Balboa Spa Client integration."""
+import time
+
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import SIGNAL_UPDATE
+
+
+class BalboaEntity(Entity):
+    """Abstract class for all Balboa platforms.
+
+    Once you connect to the spa's port, it continuously sends data (at a rate
+    of about 5 per second!).  The API updates the internal states of things
+    from this stream, and all we have to do is read the values out of the
+    accessors.
+    """
+
+    _attr_should_poll = False
+
+    def __init__(self, hass, entry, client, devtype, num=None):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.